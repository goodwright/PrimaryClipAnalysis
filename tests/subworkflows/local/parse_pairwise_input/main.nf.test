nextflow_workflow {

    name "Test Workflow PARSE_PAIRWISE_INPUT"
    script "subworkflows/local/parse_pairwise_input/main.nf"
    config "tests/subworkflows/local/parse_pairwise_input/.testconfig"
    profile "singularity"
    workflow "PARSE_PAIRWISE_INPUT"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "${projectDir}/tests/subworkflows/local/parse_pairwise_input/output"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = file("${projectDir}/tests/ARTR-seq/input_data/pairwise_peak_calling_for_testing.csv")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()

            // asseert the pairwise channel contains [WT_BRD4_plus_R1,WT_No_Ab_R1]
            with(workflow.out.pairwise) {
                assert size() == 2
                assert contains(['WT_BRD4_plus_R1', 'WT_No_Ab_R1'])
            }
        }

    }

}
