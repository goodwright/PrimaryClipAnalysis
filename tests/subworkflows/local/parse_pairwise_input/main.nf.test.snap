{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "WT_BRD4_plus_R1",
                        "WT_No_Ab_R1"
                    ],
                    [
                        "WT_BRD4_plus_R2",
                        "WT_No_Ab_R2"
                    ]
                ],
                "1": [
                    
                ],
                "pairwise": [
                    [
                        "WT_BRD4_plus_R1",
                        "WT_No_Ab_R1"
                    ],
                    [
                        "WT_BRD4_plus_R2",
                        "WT_No_Ab_R2"
                    ]
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.4"
        },
        "timestamp": "2024-10-08T14:41:40.966647"
    }
}